src/compile/selection/legends.ts

```diff
@@ -72,7 +72,7 @@ const legendBindings: SelectionCompiler<'point'> = {
           ...(!selCmpt.init ? {value: null} : {}),
           on: [
             // Legend entries do not store values, so we need to walk the scenegraph to the symbol datum.
-            {events, update: 'datum.value || item().items[0].items[0].datum.value', force: true},
+            {events, update: 'isDefined(datum.value)? datum.value: item().items[0].items[0].datum.value', force: true},
```

Comment:
Just a small formatting difference to how we usually write ternaries. 

```suggestion
            {events, update: 'isDefined(datum.value) ? datum.value : item().items[0].items[0].datum.value', force: true},
```

Reply:
Done in [ee2703f](https://github.com/vega/vega-lite/pull/8895/commits/ee2703f6386776432a0a3e4a86f2aa69f6874b84)